core/src/main/java/io/parsingdata/metal/expression/comparison/ComparisonExpression.java

```diff
@@ -57,7 +57,7 @@ public ComparisonExpression(final ValueExpression value, final ValueExpression p
 
     @Override
     public boolean eval(final ParseGraph graph, final Encoding encoding) {
-        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(Optional.of(graph.current())) : value.eval(graph, encoding);
+        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(graph.current().map(parseValue -> parseValue)) : value.eval(graph, encoding);
```

Comment:
Ahhh, generics :smile: I think an other way could be to declare the type of the values variable and compare method parameter as something like ```ImmutableList<? extends Optional<? extends Value>>```.